### ISSUE TITLE
LDAP two-way sync with users & groups

### ISSUE THREAD

**Issue by teon** [DefGuard/defguard#478]

**Comment by compgeniuses** [comment]
this is going to be an enterprise only feature?

**Comment by teon** [comment]
@compgeniuses yes - when done, this is going to be an enteprise feature - as typical small companies, homelabs, etc. in most cases does not use LDAP functionality, it's in most cases a "business" use case.

**Comment by compgeniuses** [comment]
Actually Quite to the contrary, Businesses make use of Active directory and SAML, which makes more sense for them.

I was looking into the implementation of Defguard into nethserver 8, which by default offers Ldap as a method for managing users, and since it has builtin wireguard for cluster management, now am having second thoughts.

Most IDP, just offer being able to fetch data from ldap into the IDP, in your case, your current implementation is Users are created inside Defguard, then pushed to ldap, one of the reasons why i am yet to do the impleemntation, which in my belief might give you guys alot of mileage.

Kindly consider, as this is the fundamental key Basic functionality of any IDP out there.

**Comment by teon** [comment]
@compgeniuses thank you for your response.
I can agree with SAML, but ActiveDirectory is actually LDAP.

What would you recommend (besides SAML) in terms of features to, what should be for businesses (enterprise) and what open-core?

Thank you in advance.

**Comment by compgeniuses** [comment]
Yes its noted with AD, so basically, its fine to allow 2 way sync with Ldap and AD, because actually, there are very few IDP that support 2 way sync. most support AD/Ldap to IDP for users.

I think expediting Ldap suers into Defguard should increase adoption, and i would be Happy to Also support by adding Defguard into NS8, and possibly coolify.

SAML is also important for most use cases, especially for advanced user mappings,

If you must look for an implementation to impose an Enterprise class into, that would be SCIM. its still new, and equally complex, plus extremely complex enterprise have a need for it.

I understand maintaining an OSS project is hard, and equally balancing between Enteprise and Community Features is not an easy job, however, understand that there are many IDP that already offer Alot of features you are offering, which you shoudl consider leaving them into Community Edition.

Its the Features that Most IDP dont have, and are not even looking into implementing that should Really fall into Enterprise.

So Far i only know Authentik and some commercial offerings offering SCIM, but if that's also just left for everyone, might also be compelling for many to switch to Defguard.

I have seen many items in the milestone project page, that have been marked for Enteprise use. Those have no significant impact on most users, However, this as well as [#697](https://github.com/DefGuard/defguard/issues/697) epcially this second one, i smeant to help with Initial Adoption, JUst consider.

**Comment by t-aleksander** [comment]
First iteration of this feature has been done and is available in 1.3.0 alphas

**Comment by compgeniuses** [comment]
> First iteration of this feature has been done and is available in v1.3.0 alphas. [@compgeniuses](https://github.com/compgeniuses)

wonderful news. Thank you for the Update.. Ill work on testing the same
